src/robot_state_RT.cpp

```diff
@@ -311,124 +223,96 @@ std::vector<double> RobotStateRT::getVActual() {
+        offset += sizeof(double) * (15); // TODO: double check that this offset is correct
+      }
+      else{
+        offset += sizeof(double) * (3+15); // TODO: double check that this offset is correct
+      }
+      tcp_force_ = unpackVector(buf, offset, 6);
+      tool_vector_actual_ = unpackVector(buf, offset, 6);
+      tcp_speed_actual_ = unpackVector(buf, offset, 6);
+    } else {
+      i_control_ = unpackVector(buf, offset, 6);
+      tool_vector_actual_ = unpackVector(buf, offset, 6);
+      tcp_speed_actual_ = unpackVector(buf, offset, 6);
+      tcp_force_ = unpackVector(buf, offset, 6);
+      tool_vector_target_ = unpackVector(buf, offset, 6);
+      tcp_speed_target_ = unpackVector(buf, offset, 6);
+    }
 
-	memcpy(&digital_input_bits, &buf[offset], sizeof(digital_input_bits));
-	digital_input_bits_ = unpackDigitalInputBits(be64toh(digital_input_bits));
-	offset += sizeof(double);
-	motor_temperatures_ = unpackVector(buf, offset, 6);
-	offset += sizeof(double) * 6;
-	memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-	controller_timer_ = ntohd(unpack_to);
-	if (version_ > 1.6) {
-		offset += sizeof(double) * 2;
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		robot_mode_ = ntohd(unpack_to);
-		if (version_ > 1.7) {
-			offset += sizeof(double);
-			joint_modes_ = unpackVector(buf, offset, 6);
-		}
-	}
-	if (version_ > 1.8) {
-		offset += sizeof(double) * 6;
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		safety_mode_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		tool_accelerometer_values_ = unpackVector(buf, offset, 3);
-		offset += sizeof(double) * 3;
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		speed_scaling_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		linear_momentum_norm_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		v_main_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		v_robot_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		memcpy(&unpack_to, &buf[offset], sizeof(unpack_to));
-		i_robot_ = ntohd(unpack_to);
-		offset += sizeof(double);
-		v_actual_ = unpackVector(buf, offset, 6);
-	}
-	val_lock_.unlock();
+    memcpy(&digital_input_bits, &buf[offset], sizeof(digital_input_bits));
+    digital_input_bits_ = unpackDigitalInputBits(be64toh(digital_input_bits));
+    offset += sizeof(double);
+    motor_temperatures_ = unpackVector(buf, offset, 6);
+    controller_timer_   = unpackDouble(buf, offset);
+    if (version_ > 1.6) {
+      offset += sizeof(double); // TODO: double check that this offset is correct
+      robot_mode_ = unpackDouble(buf, offset);
+      if (version_ > 1.7) {
+        joint_modes_ = unpackVector(buf, offset, 6);
+      }
+      else{
+        // TODO: double check that this offset is correct
+        offset += sizeof(double) * 6;
```

Comment:
this offset was added to replace the one at line 408 of the old code when version_ <= 1.7